src/hotspot/share/gc/shenandoah/shenandoahYoungGeneration.hpp

```diff
@@ -52,6 +52,12 @@ class ShenandoahYoungGeneration : public ShenandoahGeneration {
     return _old_gen_task_queues;
   }
 
+  // Returns true if the young generation is configured to enqueue old
+  // oops for the old generation mark queues.
+  bool is_bootstrap_cycle() {
```

Comment:
Why is this called a `bootstrap cycle`? I must be missing some big picture background of nomenclature here.

Reply:
Every old generation cycle is preceded by a young collection. We call this a bootstrap cycle because it populates the old generation mark queues with old objects it encountered during the marking of young. Otherwise, we'd have to maintain a reverse-remembered set for young->old pointers.